.github/workflows/test.yaml

```diff
@@ -264,10 +272,76 @@ jobs:
         files: .coverage/coverage-*.cov
         verbose: true
 
-  spread:
+  spread-fundamental:
     uses: ./.github/workflows/spread-tests.yaml
     needs: [unit-tests, snap-builds]
     name: "spread ${{ matrix.group }}"
+    with:
+      # Github doesn't support passing sequences as parameters.
+      # Instead here we create a json array and pass it as a string.
+      # Then in the spread workflow it turns it into a sequence 
+      # using the fromJSON expression.
+      runs-on: '["self-hosted", "spread-enabled"]'
+      group: ${{ matrix.group }}
+      backend: ${{ matrix.backend }}
+      systems: ${{ matrix.systems }}
+      tasks: ${{ matrix.tasks }}
+      rules: ${{ matrix.rules }}
+      is-fundamental: true
+    strategy:
+      # FIXME: enable fail-fast mode once spread can cancel an executing job.
+      # Disable fail-fast mode as it doesn't function with spread. It seems
+      # that cancelling tasks requires short, interruptible actions and
+      # interrupting spread, notably, does not work today. As such disable
+      # fail-fast while we tackle that problem upstream.
+      fail-fast: false
+      matrix:
+        include:
+          - group: 'debian-req (fundamental)'
+            backend: google-distro-1
+            systems: 'debian-11-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'fedora (fundamental)'
+            backend: openstack
+            systems: 'fedora-40-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-focal (fundamental)'
+            backend: google
+            systems: 'ubuntu-20.04-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-noble (fundamental)'
+            backend: google
+            systems: 'ubuntu-24.04-64'
+            tasks: 'tests/...'
+            rules: 'main'
+          - group: 'ubuntu-dev-previous (fundamental)'
+            backend: google
+            systems: 'ubuntu-24.10-64'
+            tasks: 'tests/...'
+            rules: 'main'
```

Comment:
As mentioned below, I would prefer `ubuntu-interim` or `ubuntu-current`, as this is the current Ubuntu release. When an LTS is the most current, this could either be the previous non-LTS (e.g. 25.10 just after 26.04 comes out), in which case it should IMO be called `ubuntu-interim`, or it could be empty for that cycle (until 26.10 comes out, for example).